src/main/java/org/commcare/cases/entity/NodeEntityFactory.java

```diff
@@ -73,10 +73,7 @@ public Detail getDetail() {
                  * We are wrapping the original exception in a new XPathException to avoid
                  * refactoring large number of functions caused by throwing XPathSyntaxException here.
                  */
-                XPathException xe = new XPathException(e.getMessage(), e);
-                if (e instanceof XPathException) {
-                    xe.setSource(((XPathException)e).getSource());
-                }
+                XPathException xe = new XPathException(e);

```

Comment:
For the same reason as that of the change you made in the android part of this PR, don't you only want to create this wrapped exception if the original exception was an `XPathSyntaxException`? Otherwise you have the same issue of wrapping an `XPathSyntaxException` by another `XPathSyntaxException`